lib/utils/configuration.loader.ts

```diff
@@ -1,38 +1,61 @@
-import { readFileSync } from 'node:fs';
+import { closeSync, existsSync, openSync, readFileSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
 
-import * as Joi from 'joi';
+import { Answers } from 'inquirer';
 
+import { MESSAGES } from '../../lib/ui/messages.js';
+import { deployConfigSchema } from '../validation/configs.schema.js';
+
+export interface SSHConfiguration {
+  username: string;
+  hostname: string;
+  pathToKeyFile: string;
+}
+
+export interface BuildConfiguration {
+  server: 'nginx';
+  nodeVersion: string;
+}
 export interface DeployConfiguration {
-  ssh: {
-    username: string;
-    hostname: string;
-    pathToKeyFile: string;
-  };
-  build: {
-    // Only nginx is supported currently
-    server: 'nginx';
-    nodeVersion: string;
-  };
+  ssh: SSHConfiguration;
+  build: BuildConfiguration;
 }
 
 export class ConfigurationLoader {
-  public static validationSchema = Joi.object<DeployConfiguration>({
-    ssh: Joi.object({
-      username: Joi.string().required(),
-      hostname: Joi.string().required(),
-      pathToKeyFile: Joi.string().required(),
-    }).required(),
-  });
+  public static configFileName = '.dsdeployrc';
+
+  public static validationSchema = deployConfigSchema;
+
+  public static getConfigPath() {
+    const configFilePath = join(process.cwd(), this.configFileName);
+    return configFilePath;
+  }
 
   public static async load(configFilePath: string) {
     try {
       const buffer = readFileSync(configFilePath);
       const configFile = JSON.parse(buffer.toString()) as DeployConfiguration;
 
       const values = await this.validationSchema.validateAsync(configFile);
+
       return values;
-    } catch {
+    } catch (e) {
+      console.error(MESSAGES.INVALID_CONFIGS);
+
       return null;
     }
   }
+
+  public static upsert(key: string, values: Answers) {
+    const configFilePath = this.getConfigPath();
+    if (!existsSync(configFilePath)) {
+      closeSync(openSync(configFilePath, 'w'));
```

Comment:
you can just assign an empty string to `configString` variable instead of performing 2 IO operations.